cmd/argocd-k8s-auth/commands/aws.go

```diff
@@ -62,6 +54,41 @@ func newAWSCommand() *cobra.Command {
 	return command
 }
 
+func getSignedRequestWithRetry(timeout time.Duration, clusterName, roleARN string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	for {
+		signed, err := getSignedRequest(clusterName, roleARN)
+		if err == nil {
+			return signed, nil
```

Comment:
Want to add a log line for err == nil saying what the error was and that you're going to retry?

Reply:
Would love to but I guess we shouldn't add anything to stdout in this particular case as it will be used by the fork exec process:
https://github.com/argoproj/argo-cd/blob/f3140fd1ecfd3cefe8835d0392a8238864ae7169/cmd/argocd-k8s-auth/commands/aws.go#L49

Reply:
Makes sense! I guess we could push it to sterr, but I'm not too worried.